File: internal/api/pages.go
package api

// SSR page discriminators embedded in #initial-data (PAT-004).
// Values must stay identical to the OpenAPI SSRPage enum and web/src/lib/pages.ts.
const (
	PageHome  = "home"
	PageClub  = "club"
	PageMatch = "match"
)

// AllSSRPages lists every valid SSR page ID in stable order.
func AllSSRPages() []string {
	return []string{PageHome, PageClub, PageMatch}
}

// ValidSSRPage reports whether page is a known SSR discriminator.
func ValidSSRPage(page string) bool {
	switch page {
	case PageHome, PageClub, PageMatch:
		return true
	default:
		return false
	}
}

File: web/src/lib/pages.ts
import type { components } from "../api/generated/schema";

/** OpenAPI SSRPage discriminator for PAT-004 #initial-data payloads. */
export type SSRPage = components["schemas"]["SSRPage"];

/**
 * Canonical SSR page IDs. Values must stay identical to:
 * - OpenAPI `components.schemas.SSRPage`
 * - `internal/api` page constants
 */
export const SSR_PAGE = {
  home: "home",
  club: "club",
  match: "match",
} as const satisfies { [K in SSRPage]: K };
